storage/kafka/src/main/java/io/apicurio/registry/kafka/snapshot/StorageSnapshotSerde.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2019 Red Hat
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.apicurio.registry.kafka.snapshot;
+
+import io.apicurio.registry.kafka.utils.SelfSerde;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutput;
+import java.io.ObjectOutputStream;
+
+/**
+ * @author Ales Justin
+ */
+public class StorageSnapshotSerde extends SelfSerde<StorageSnapshot> {
+
+    // TODO -- optimize this if/when need (via Protobuf)
+
+    @Override
+    public StorageSnapshot deserialize(String topic, byte[] data) {
+        try {
+            ByteArrayInputStream bais = new ByteArrayInputStream(data);
+            ObjectInput in = new ObjectInputStream(bais);
+            return (StorageSnapshot) in.readObject();
+        } catch (Exception e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    @Override
+    public byte[] serialize(String topic, StorageSnapshot data) {
```

Comment:
I wonder how scalable this end up being. Kafka is fine with smallish messages (tens of MB). But you're at the mercy of the Kafka admin and how they've configured `max.request.size`. I wonder how big the serialized form is if there are, say, a few thousand schemas.

Reply:
What more can we do?
I can try and re-write this to Protobuf ... but even then it can get big ...

Reply:
I don't think you have to do anything immediately, but one common technique is to fragment the data across multiple Kafka records, and reassemble it in the consumer. To do that now would be premature, but it's worth having an idea about how you might be able to evolve the serialization format in order to cope with a problem like this should it become an issue.